### PR TITLE
Add filtering examples

### PR DESCRIPTION
**Pull request by philkra**
Closes #

## Summary

Adds two more complex examples to highlight how to combine filter operators.

### Timing

**Release**:

- [x] When ready

### CODE REVIEW

**Review by kostasb — APPROVED** [review]

File: 040-SDK/115-filtering.mdx [hunk 1]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+255.231.52.33,27/Dec/2037:12:00:00 +0530,PUT,usr/admin/developer,HTTP/1.0,403,5054,http://www.parker-miller.org/tag/list/list/privacy/,Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML; like Gecko) Chrome/87.0.4280.88 Safari/537.36 OPR/73.0.3856.329,3629
+238.217.83.154,27/Dec/2037:12:00:00 +0530,DELETE,usr/register,HTTP/1.0,304,5152,-,Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML; like Gecko) Chrome/89.0.4380.0 Safari/537.36 Edg/89.0.759.0,4679
+119.170.1.203,27/Dec/2037:12:00:00 +0530,GET,usr,HTTP/1.0,303,5011,http://www.parker-miller.org/tag/list/list/privacy/,Mozilla/5.0 (Linux; Android 10; ONEPLUS A6000) AppleWebKit/537.36 (KHTML; like Gecko) Chrome/87.0.4280.141 Mobile Safari/537.36,4541
+59.107.116.6,27/Dec/2037:12:00:00 +0530,POST,usr,HTTP/1.0,502,5008,-,Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.75.14 (KHTML; like Gecko) Version/7.0.3 Safari/7046A194A,1213
+```
+
+The data source is [kaggle](https://www.kaggle.com/datasets/vishnu0399/server-logs/data?select=logfiles.log) and licensed as [public domain](https://creativecommons.org/publicdomain/zero/1.0/).
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1287]:
```suggestion
The dataset, available on [Kaggle](https://www.kaggle.com/datasets/vishnu0399/server-logs/data?select=logfiles.log), consists of server logs and is licensed under the [public domain](https://creativecommons.org/publicdomain/zero/1.0/).
```

File: 040-SDK/115-filtering.mdx [hunk 2]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+For illustration purposes, the data has been trimmed to 1000 records and converted to a CSV for a convenient [import](/docs/csv-data/import-data). The file can be downloaded [here]().
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1288]: I guess a link is needed for `here` as well...

```suggestion
In this example, the data has been trimmed to 1000 records and converted to a CSV for a convenient [import](/docs/csv-data/import-data). The file can be downloaded [here]().
```

File: 040-SDK/115-filtering.mdx [hunk 3]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
 ```
 
 </TabbedCode>
+
+## Examples
+
+We will walk through various examples to demonstrate the use cases of the different filtering options and operators.
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1271]:
```suggestion
The examples provided illustrate how to use various filtering options and operators in different use cases.
```

File: 040-SDK/115-filtering.mdx [hunk 4]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+For the example we use an access log of a web server, saved in a table called `access_logs`.
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1272]:
```suggestion
In this example, use the web server's access log, which is stored in a table named `access_logs`.
```

File: 040-SDK/115-filtering.mdx [hunk 5]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+119.170.1.203,27/Dec/2037:12:00:00 +0530,GET,usr,HTTP/1.0,303,5011,http://www.parker-miller.org/tag/list/list/privacy/,Mozilla/5.0 (Linux; Android 10; ONEPLUS A6000) AppleWebKit/537.36 (KHTML; like Gecko) Chrome/87.0.4280.141 Mobile Safari/537.36,4541
+59.107.116.6,27/Dec/2037:12:00:00 +0530,POST,usr,HTTP/1.0,502,5008,-,Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.75.14 (KHTML; like Gecko) Version/7.0.3 Safari/7046A194A,1213
+```
+
+The data source is [kaggle](https://www.kaggle.com/datasets/vishnu0399/server-logs/data?select=logfiles.log) and licensed as [public domain](https://creativecommons.org/publicdomain/zero/1.0/).
+For illustration purposes, the data has been trimmed to 1000 records and converted to a CSV for a convenient [import](/docs/csv-data/import-data). The file can be downloaded [here]().
+
+### Multiple exact matches
+
+In the following query we want to fetch all records that have the status_code exactly `502` and the HTTP method `DELETE`.
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1292]:
```suggestion
In the upcoming query, you'll retrieve all records that have a status_code of exactly `502` and use the HTTP method `DELETE`.
```

File: 040-SDK/115-filtering.mdx [hunk 6]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+In order to combine both filter conditions, we must use the `$all` operator.
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1293]:
```suggestion
To combine both filter conditions, you must use the `$all` operator.
```

File: 040-SDK/115-filtering.mdx [hunk 7]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+records = xata.data().query("access_logs", {
+  "filter": {
+    "$all": [
+        { # IP range starts with `202.8.*` or `29.171.152.*`
+          "$any": [
+            { "ip": { "$startsWith": "202.8." } },
+            { "ip": { "$startsWith": "29.171.152." } },
+          ]
+        },
+        # attempted to login with PUT or POST
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1381]: I know these are just the comments, but still

File: 040-SDK/115-filtering.mdx [hunk 8]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+        { "method": { "$any": ["PUT", "POST"] } },
+        # login url_path exact match
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1383]:
```suggestion
        # Log in url_path exact match
```

File: 040-SDK/115-filtering.mdx [hunk 9]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+        { "url_path": { "$is": "usr/login" } },
+        { # did not produce the status code `200`
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1385]:
```suggestion
        { # Did not produce the status code `200`
```

File: 040-SDK/115-filtering.mdx [hunk 10]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+
+```ts
+const records = await xata.db.access_logs
+  .filter({
+    $all: [
+      {
+        // IP range starts with `202.8.*` or `29.171.152.*`
+        $any: [{ ip: { $startsWith: '202.8.' } }, { ip: { $startsWith: '29.171.152.' } }]
+      },
+      // attempted to login with PUT or POST
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1356]:
```suggestion
      // Attempted to log in with PUT or POST
```

File: 040-SDK/115-filtering.mdx [hunk 11]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+      { method: { $any: ['PUT', 'POST'] } },
+      // login url_path exact match
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1358]:
```suggestion
      // Log in url_path exact match
```

File: 040-SDK/115-filtering.mdx [hunk 12]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+      { url_path: { $is: 'usr/login' } },
+      {
+        // did not produce the status code `200`
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1361]:
```suggestion
        // Did not produce the status code 200
```

File: 040-SDK/115-filtering.mdx [hunk 13]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+- did not produce the status code `200`
+
+<TabbedCode tabs={['TypeScript', 'Python', 'SQL', 'JSON']}>
+
+```ts
+const records = await xata.db.access_logs
+  .filter({
+    $all: [
+      {
+        // IP range starts with `202.8.*` or `29.171.152.*`
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1353]: I dont think the back ticks will appear in the code snippet

```suggestion
        // IP range starts with 202.8.* or 29.171.152.*
```

File: 040-SDK/115-filtering.mdx [hunk 14]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+  }
+}
+```
+
+</TabbedCode>
+
+### Nested operators
+
+This example will demonstrate how to use multiple operators.
+We want to get all records that:
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1339]:
```suggestion
You want to get all records that:
```

File: 040-SDK/115-filtering.mdx [hunk 15]
```diff
@@ -1245,3 +1265,161 @@ records = xata.data().query("Posts", {
...
+
+- IP range starts with `202.8.*` or `29.171.152.*`
+- attempted to login on URL path `usr/login`
+- HTTP methods are `PUT` or `POST`
+- did not produce the status code `200`
```
Comment (joan-ing) [inline 040-SDK/115-filtering.mdx:1344]:
```suggestion
- Did not produce the status code `200`
```

**Pull request merged**